### PR TITLE
Add onError prop to handle broken urls and failed downloads

### PR DESCRIPTION
**Pull request by MindFreeze**
Addresses issue #89

### CODE REVIEW

**Comment by MindFreeze** [comment]
Is anyone going to merge this or comment on it at least?

File: src/Image.js [hunk 1]
```diff
@@ -29,19 +30,28 @@ export default class Image extends React.Component<ImageProps, ImageState> {
...
     };
 
-    async load({uri, options = {}}: ImageProps): Promise<void> {
+    async load({uri, options = {}, onError}: ImageProps): Promise<void> {
         if (uri) {
-            const path = await CacheManager.get(uri, options).getPath();
-            if (this.mounted) {
-                this.setState({ uri: path });
+            try {
+                const path = await CacheManager.get(uri, options).getPath();
```
Comment (gabmichels) [inline src/Image.js]: I would consider throwing an Error here: https://github.com/wcandillon/react-native-expo-image-cache/blob/e734867fc8bb386b02ce1e1eafc0d5cbe42288ba/src/CacheManager.js#L33
 in order for your catch to get triggered

File: src/Image.js [hunk 2]
```diff
@@ -29,19 +30,28 @@ export default class Image extends React.Component<ImageProps, ImageState> {
...
+                if (this.mounted) {
+                    if (path) {
+                        this.setState({ uri: path });
+                    } else {
+                        onError(new Error("Could not load image"));
```
Comment (wcandillon) [inline src/Image.js]: Here the error format should with symetric with onError from `<Image>` no?

Comment (MindFreeze) [reply src/Image.js]: I updated it to use the same format

**Comment by MindFreeze** [comment]
Can someone merge this before it goes stale?

**Pull request merged**